docs/en/index.rst

```diff
@@ -323,6 +327,153 @@ With this configuration in place, any time you send an email with the ``default`
+            }
+
+            $startTime = microtime(true) * 1000;
+            $this->dispatchEvent('Processor.message.start', ['message' => $jobMessage]);
+
+            try {
+                $response = $this->processMessage($jobMessage);
+            } catch (Throwable $e) {
+                $message->setProperty('jobException', $e);
+
+                $this->logger->debug(sprintf('Message encountered exception: %s', $e->getMessage()));
+                $this->dispatchEvent('Processor.message.exception', [
+                    'message' => $jobMessage,
+                    'exception' => $e,
+                    'duration' => (int)((microtime(true) * 1000) - $startTime),
+                ]);
+
+                return Result::requeue('Exception occurred while processing message');
+            }
+
+            $duration = (int)((microtime(true) * 1000) - $startTime);
+
+            if ($response === InteropProcessor::ACK) {
+                $this->logger->debug('Message processed successfully');
+                $this->dispatchEvent('Processor.message.success', [
+                    'message' => $jobMessage,
+                    'duration' => $duration,
+                ]);
+
+                return InteropProcessor::ACK;
+            }
+
+            if ($response === InteropProcessor::REJECT) {
+                $this->logger->debug('Message processed with rejection');
+                $this->dispatchEvent('Processor.message.reject', [
+                    'message' => $jobMessage,
+                    'duration' => $duration,
+                ]);
+
+                return InteropProcessor::REJECT;
+            }
+
+            $this->logger->debug('Message processed with failure, requeuing');
+            $this->dispatchEvent('Processor.message.failure', [
+                'message' => $jobMessage,
+                'duration' => $duration,
+            ]);
+
+            return InteropProcessor::REQUEUE;
+        }
+    }
+
+Configuration example::
+
+    'Queue' => [
+        'default' => [
+            'url' => 'redis://localhost:6379',
+            'queue' => 'default',
+            // No processor specified - uses default Processor class
+        ],
+        'timed' => [
+            'url' => 'redis://localhost:6379',
+            'queue' => 'timed',
+            'processor' => \App\Queue\TimedProcessor::class, // Custom processor with timing
+        ],
+    ],
+
+**Note**: If no processor is specified in the configuration, the default
+``Cake\Queue\Queue\Processor`` class will be used. Custom processors are useful
+for adding custom logging, metrics collection, or specialized message handling.
+
+**Important**: The `--processor` command line option is different from the `processor` configuration option:
+
+- **Configuration `processor`**: Specifies the processor class to use for processing messages
+- **Command line `--processor`**: Specifies the processor name for Enqueue topic binding (used in `bindTopic()`)
```

Comment:
Should we rename the CLI flag to `processor-name` (can be done separately)? The current overlap is awkward.